sklearn/base.py

```diff
@@ -292,6 +293,24 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        if type(self).__module__.startswith('sklearn.'):
+            return dict(self.__dict__.items(), __version__=__version__)
+        else:
+            return dict(self.__dict__.items())
```

Comment:
Can we still have that only on models inside scikit-learn?  It would be a class attribute, right, so we wouldn't have access to ``self``.
We can't set it in ``__init__`` unless we add ``super`` calls to all our ``__init__``s (which I'd try to avoid) or using meta-classes (which I'd try to avoid even more).